src/components/screens/vacation-requests/vacation-requests-screen.tsx

```diff
@@ -0,0 +1,287 @@
+    if (!userProfile || !userProfile.id) return;
+
+    try {
+      if (createdRequest.id) {
+        const createdStatus = await vacationRequestStatusApi.createVacationRequestStatus({
+          id: createdRequest.id,
+          vacationRequestStatus: {
+            vacationRequestId: createdRequest.id,
+            status: VacationRequestStatuses.PENDING,
+            message: createdRequest.message,
+            createdAt: new Date(),
+            createdBy: userProfile.id,
+            updatedAt: new Date(),
+            updatedBy: userProfile.id
+          }
+        });
+
+        setVacationRequestStatuses([createdStatus, ...vacationRequestStatuses]);
+      }
+    } catch (error) {
+      setError(`${strings.vacationRequestError.createStatusError}, ${error}`);
+    }
+  };
+
+  /**
+   * Create a vacation request
+   *
+   * @param vacationData vacation data
+   */
+  const createVacationRequest = async (vacationData: VacationData) => {
+    if (!userProfile || !userProfile.id) return;
+
+    try {
+      const createdRequest = await vacationRequestsApi.createVacationRequest({
+        vacationRequest: {
+          personId: userProfile.id,
+          createdBy: userProfile.id,
+          startDate: vacationData.startDate.toJSDate(),
+          endDate: vacationData.endDate.toJSDate(),
+          type: vacationData.type,
+          message: vacationData.message,
+          createdAt: new Date(),
+          updatedAt: new Date(),
+          days: vacationData.days
+        }
+      });
+
+      createVacationRequestStatus(createdRequest);
+      setVacationRequests([createdRequest, ...vacationRequests]);
+    } catch (error) {
+      setError(`${strings.vacationRequestError.createRequestError}, ${error}`);
+    }
+  };
+
+  /**
+   * Update a vacation request
+   *
+   * @param vacationData vacation request data
+   * @param vacationRequestId vacation request id
+   */
+  const updateVacationRequest = async (vacationData: VacationData, vacationRequestId: string) => {
+    if (!userProfile || !userProfile.id) return;
+
+    try {
+      const vacationRequest = vacationRequests.find(
+        (vacationRequest) => vacationRequest.id === vacationRequestId
+      );
+      if (vacationRequest) {
+        const updatedRequest = await vacationRequestsApi.updateVacationRequest({
+          id: vacationRequestId,
+          vacationRequest: {
+            ...vacationRequest,
+            startDate: vacationData.startDate.toJSDate(),
+            endDate: vacationData.endDate.toJSDate(),
+            type: vacationData.type,
+            message: vacationData.message,
+            updatedAt: new Date(),
+            days: vacationData.days
+          }
+        });
+        const updatedVacationRequests = vacationRequests.map((vacationRequest) => {
+          if (vacationRequest.id === updatedRequest.id) {
+            return updatedRequest;
+          }
+          return vacationRequest;
```

Comment:
Could use ternary here e.g. vacationRequest.id === updatedRequest.id ? updatedRequest : vacationRequest. This could be done on a single line without the curly brackets of the map

Reply:
Nice. This works too and looks better